staging/src/k8s.io/apiserver/pkg/admission/plugin/validatingadmissionpolicy/controller.go

```diff
@@ -298,38 +306,136 @@ func (c *celAdmissionController) Validate(
 				timeoutCtx, cancel := context.WithTimeout(c.policyController.context, 1*time.Second)
 				defer cancel()
 
-				if !cache.WaitForCacheSync(timeoutCtx.Done(), paramController.HasSynced) {
+				if !cache.WaitForCacheSync(timeoutCtx.Done(), definitionInfo.paramController.HasSynced) {
 					addConfigError(fmt.Errorf("paramKind kind `%v` not yet synced to use for admission",
 						paramKind.String()), definition, binding)
 					continue
 				}
+			}
+
+			// Find params to use with policy
+			switch {
+			case paramKind == nil:
+				// ParamKind is unset. Ignore any globalParamRef or namespaceParamRef
+				// setting.
+				// TODO(alexzielenski): historical behavior. Should the failure mode apply here instead?
+				//
```

Comment:
I'm in favor of keeping this as-is.

My main consideration is that when policy authors make changes to their resources that they have safe ways to roll out the changes.

Ignoring paramRef when paramKind is unset makes it possible to add params to resources and remove params from resources by enabling these workflows to happen safely:

- (1) Add params resources (2) Add params to bindings (3) add params to policy
- (1) Remove params from a policy (2) remove params from bindings (3) remove param resources

If we allow bindings to failure in these workflows, these change rollout paths become unsafe..

Reply:
removed TODO